### DISCUSSION TITLE
Support for Azure OpenAI

### DISCUSSION THREAD

**Discussion by Giustino98** [parsee-ai/parsee-core#7] [Q&A] [answered]
Hi,

does the library current support different LLM models? From what I've seen from tutorials, I have seen support for replicate models and openai, but not for all the others (e.g. Azure OpenAI). Am I missing anything from docs or it is a feature in the roadmap?

Thank you so much.

**Comment by thf24** [comment, marked as answer]
Hi,
in this folder you can see the models that are supported right now: extraction/models/llm_models
so currently there is support for:
- openai
- anthropic
- replicate
- together AI
- mistral APIs (e.g. Mistral Large)
- cohere
- Ollama (with any host of course, so not just locally)

It is also quite easy to add support for a new endpoint, or you can also do it yourself (personally I haven't tried using Azure for LLMs yet).

For that you would basically have to do two things:
- write a new class that implements LLMBaseModel (you can use all the files in extraction/models/llm_models as a template for that, e.g. extraction/models/llm_models/chatgpt_model.py), there you would basically configure how to call the azure endpoint etc. Let's call this "MyCustomLLMBaseModel"
- write a new class that extends the ModelLoader (parsee/extraction/models/model_loader.py). You can then overwrite the get_question_model_method to return your custom LLMBaseModel instance, something like this:
```
class CustomModelLoader(ModelLoader):

    def get_question_model(self, model_name: Optional[str], items: List[GeneralQueryItemSchema], all_meta_items: List[StructuringItemSchema], params: Dict[str, Any]) -> Union[QuestionModel, None]:
            return LLMQuestionModel(items, all_meta_items, self.storage,MyCustomLLMBaseModel(), **params)
```

The custom ModelLoader you can then pass to the run_job_with_single_model method for example.
